spec/requests/api/shared/shared_examples.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+RSpec.shared_examples "a JSON endpoint" do |expected_response_status|
+  before do |example|
+    submit_request(example.metadata)
+  end
+
+  describe "response status" do
+    it "returns expected response status" do
+      expect(response.status).to eq(expected_response_status)
+    end
+  end
+
+  describe "request body" do
+    it "matches the documented request schema" do |example|
+      schemer = JSONSchemer.schema(expected_request_schema.schemer)
+      valid = schemer.valid?(params)
+      unless valid # for debugging
```

Comment:
Did you mean to leave this `unless valid` part in? If you did maybe making a helper method would be useful e.g. `expect_schema_valid(schemer, params)` that does the expectation and also the puts debugging on failure.

Reply:
Yea its handy to have when trying to figure out why a spec failed, but yes I'll work on a helper method for this